controllers/apply-next/validators.js

```diff
@@ -1,31 +1,91 @@
 'use strict';
 const moment = require('moment');
 const baseJoi = require('joi');
-const { toInteger, isObject } = require('lodash');
+const { isEmpty, isArray, filter, toInteger, isObject } = require('lodash');
+
 const { POSTCODE_REGEX } = require('../../modules/postcodes');
 
-const Joi = baseJoi.extend(joi => ({
-    base: joi.date(),
-    name: 'dateObject',
-    coerce: function(value, state, options) {
-        if (isObject(value)) {
-            const date = moment({
-                year: toInteger(value.year),
-                // month is 0 indexed when constructing a date object
-                month: toInteger(value.month) - 1,
-                day: toInteger(value.day)
-            });
-
-            if (date.isValid()) {
-                return date.toISOString();
+const dateObject = joi => {
+    return {
+        base: joi.date(),
+        name: 'dateObject',
+        coerce: function(value, state, options) {
+            if (isObject(value)) {
+                const date = moment({
+                    year: toInteger(value.year),
+                    // month is 0 indexed when constructing a date object
+                    month: toInteger(value.month) - 1,
+                    day: toInteger(value.day)
+                });
+
+                if (date.isValid()) {
+                    return date.toISOString();
+                } else {
+                    return this.createError('date.isoDate', { v: value }, state, options);
+                }
             } else {
-                return this.createError('date.isoDate', { v: value }, state, options);
+                return value;
             }
-        } else {
-            return value;
         }
-    }
-}));
+    };
+};
+
+const budgetValidator = joi => {
+    return {
+        base: joi.array(),
+        name: 'budgetItems',
+        coerce(value, state, options) {
+            if (isArray(value)) {
+                // Strip out anything that doesn't have an item name and a cost
+                // (eg. validate things that are half-supplied, but not empty)
+                return filter(value, line => !isEmpty(line.item) || !isEmpty(line.cost));
+            } else {
+                return value;
+            }
+        },
+        pre(value, state, options) {
+            if (this._flags.maxBudget) {
+                const total = value.reduce((acc, cur) => acc + cur.cost, 0);
+                if (total > this._flags.maxBudget) {
+                    return this.createError(
+                        'budgetItems.overBudget',
+                        { v: value, number: this._flags.maxBudget },
+                        state,
+                        options
+                    );
+                }
+            }
+            return value;
+        },
+        language: {
+            maxBudget: 'needs to be a number'
```

Comment:
This error gets shown in your console (eg. when developing), not in the browser, FYI.